zerver/views/auth.py

```diff
@@ -393,9 +390,8 @@ def finish_google_oauth2(request: HttpRequest) -> HttpResponse:
     if mobile_flow_otp is not None:
         # When request was not initiated from subdomain.
         user_profile, return_data = authenticate_remote_user(realm, email_address)
-        invalid_subdomain = bool(return_data.get('invalid_subdomain'))
-        return login_or_register_remote_user(request, email_address, user_profile,
```

Comment:
`authenticate_remote_user()` always returns `return_data = {}`. So `invalid_subdomain` is always `False`